File: src/main/java/io/nightfis/createstellarexoduscore/filter/IsEntityIdFilter.java
package io.nightfis.createstellarexoduscore.filter;

import net.minecraft.network.chat.Component;
import net.minecraft.world.entity.LivingEntity;

public class IsEntityIdFilter extends TargetFilter {

    private static final String ID = "IsEntityIdFilter";

    @Override
    public String getId() {
        return ID;
    }

    @Override
    public boolean match(LivingEntity entity, String arg) {
        return entity.getType().getDescriptionId().equals(arg);
    }

    @Override
    public boolean needArg() {
        return true;
    }

    @Override
    public Component title() {
        return Component.translatable("screen.createstellarexoduscore.auto_aim_turret.filter.is_entity_id");
    }

    @Override
    public String tooltip(String arg) {
        return Component.translatable(arg).getString();
    }
}

File: src/main/kotlin/io/nightfis/createstellarexoduscore/filter/IsEntityIdFilter.kt
package io.nightfis.createstellarexoduscore.filter

import net.minecraft.network.chat.Component
import net.minecraft.world.entity.LivingEntity

class IsEntityIdFilter : TargetFilter() {

    override fun getId(): String = ID

    override fun match(entity: LivingEntity, arg: String): Boolean {
        return entity.type.descriptionId == arg
    }

    override fun needArg(): Boolean = true

    override fun title(): Component {
        return Component.translatable("screen.create_stellar_exodus_core.auto_aim_turret.filter.is_entity_id")
    }

    override fun tooltip(arg: String): String {
        return Component.translatable(arg).string
    }

    companion object {
        private const val ID = "IsEntityIdFilter"
    }
}
